### PR TITLE
Blog: FLAME GPU 2 H100/A100/V100 Benchmarking

### PR DESCRIPTION
**Pull request by ptheywood**
A Blog post on the benchmarking the new H100 GPU Nodes in Stanage using a FLAME GPU 2 test case, carried out during pilot H100 access. 
This is timed to co-incide with public H100 availabililty, and will be referenced in an email from RIT when H100s become part of the default GPU pool.

Data / plotting from: https://github.com/FLAMEGPU/FLAMEGPU2-circles-benchmark/pull/21

### CODE REVIEW

**Comment by willfurnass** [comment]
cc @Carldkennedy @nikishe

File: _posts/2023-07-18-benchmarking-flamegpu2-on-h100-a100-and-v100-gpus.md [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
++ [Bessemer][bessemer] (Tier 3, The University of Sheffield):
+  + [4 public][bessemer-gpus] NVIDIA Tesla V100 SXM2 32GB GPUs
+  + [32 private (Computer Science)][bessemer-dcs-gpus] NVIDIA Tesla V100 SXM2 32GB GPUS
++ [JADE 2][jade2] (Tier 2 - Machine Learning and Molecular Dynamics):
+  + 504 NVIDIA Tesla V100 MAXQ 32GB GPUs
++ [N8 CIR Bede][bede] (Tier 2 - PPC64LE CPUs & Multi-node jobs):
+  + 136 NVIDIA Tesla V100 SXM2 32GB GPUs with host-device NVLink
+  + 16 NVIDIA Tesla T4 16GB PCIe GPUs
+
+The newly available H100 GPUs in Stanage are the [300W PCIe variant][nvidia-h100-architecture], with 2 GPUs per node which are not connected to each other by NVLink connections.
```
Comment (willfurnass) [inline _posts/2023-07-18-benchmarking-flamegpu2-on-h100-a100-and-v100-gpus.md]: Omit "with 2 GPUs per node which are not connected to each other by NVLink connection" and expand in following para as that covers NVLink specifically?  Also, worth mentioning SXM vs PCIe?

**Review by willfurnass — APPROVED** [review]
Great post!  Good that H100s offer real perf improvements for some workloads.

LTGM but see minor suggestion.  Also, do you explicitly say that all benchmarks were with single GPUs?

**Comment by EdwinB12** [comment]
Looks great to me Pete. Lovely Visuals and blog structure is clear, with a clear conclusion. Clicked on about 90% of the hyperlinks and they were all good. 

I will leave a critic of the technical material to someone more qualified but all good from me.

**Comment by willfurnass** [comment]
I second @EdwinB12's comments re the plots and other visualisations being v. handsome!

**Review by willfurnass — APPROVED** [review]

File: _posts/2023-07-18-benchmarking-flamegpu2-on-h100-a100-and-v100-gpus.md (hunk 1 above)
Comment (ptheywood) [reply _posts/2023-07-18-benchmarking-flamegpu2-on-h100-a100-and-v100-gpus.md]: Ta. 
I've now moved that to the following paragraph and expanded it a bit.

I considered adding raw numbers, but it gets complicated (64GB/s (ish) aggregate bi-directional PCI for 16 lanes of gen4, vs 600GB/s total aggregate per A100 (12 lanes at 25GB/s each way), but only 200GBs between each pair of devices, so actually its 32GB/s and 100GB/s one-way that is interesting. Plus the H100 are PCIE Gen 5, but they are connected to a Gen 4 host... 

I've also added a sentence clarifying the benchamrks are single-gpu. (but I didn't go into the fact its possibel to do ensembles of simulations which does use mutliple gpus in flamegpu, nor the fact we have an open PR about distribtued multi-node ensembles...)

File: _posts/2023-07-18-benchmarking-flamegpu2-on-h100-a100-and-v100-gpus.md [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+  + [32 private (Computer Science)][bessemer-dcs-gpus] NVIDIA Tesla V100 SXM2 32GB GPUS
++ [JADE 2][jade2] (Tier 2 - Machine Learning and Molecular Dynamics):
+  + 504 NVIDIA Tesla V100 MAXQ 32GB GPUs
++ [N8 CIR Bede][bede] (Tier 2 - PPC64LE CPUs & Multi-node jobs):
+  + 136 NVIDIA Tesla V100 SXM2 32GB GPUs with host-device NVLink
+  + 16 NVIDIA Tesla T4 16GB PCIe GPUs
+
+The newly available H100 GPUs in Stanage are the [300W PCIe variant][nvidia-h100-architecture], this means that for some workloads the university's 500W A100 SXM4 GPUs may offer higher performance (with higher power consumption).
+
+Multi-GPU workloads which perform a large volume of GPU to GPU communication may also be better suited to the A100 nodes than the H100 nodes.
```
Comment (mondus) [inline _posts/2023-07-18-benchmarking-flamegpu2-on-h100-a100-and-v100-gpus.md]:
```suggestion
For example, multi-GPU workloads which perform a large volume of GPU to GPU communication may be better suited to the A100 nodes than the H100 nodes.
```

Also move this up to follow previous sentence.

File: _posts/2023-07-18-benchmarking-flamegpu2-on-h100-a100-and-v100-gpus.md [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+## FLAME GPU 2
+
+<!-- Within the RSE group, a number of staff have been involved in developing [FLAME GPU 2][flamegpu-website]... -->
+
+[![FLAME GPU 2 Logo](/assets/images/2023-07-18-benchmarking-flamegpu2-on-h100-a100-and-v100-gpus/flamegpu2-icon-256.png){: .img-fluid max-width="256px "}][flamegpu-website]{: .float-right .ml-2}
+
+[FLAME GPU 2][flamegpu2-repo] is an open-source GPU accelerated simulator for domain independent complex systems simulations using an agent-based modelling approach.
+Models are implemented using CUDA C++ or Python 3, with modellers describing the behaviours of individuals within the simulation and how they interact with one another through message lists.
+From these relatively simple agent behaviours and interactions, complex behaviours can emerge and be observed.
+The underlying use of GPUs allows for much larger scale Agent Based Simulations than traditional CPU-based frameworks with high levels of performance, while abstracting the complexities of GPU programming away from the modeller.
```
Comment (mondus) [inline _posts/2023-07-18-benchmarking-flamegpu2-on-h100-a100-and-v100-gpus.md:56]:
```suggestion
The underlying use of GPUs allows for much larger scale Agent Based Simulations than traditional CPU-based frameworks with high levels of performance, while abstracting the complexities of GPU programming away from the modeller. A recent [NVIDIA blog](https://developer.nvidia.com/blog/fast-large-scale-agent-based-simulations-on-nvidia-gpus-with-flame-gpu/) post showed how, in certain scenarios, FLAME GPU is able to operate thousands of times faster than CPU alternatives.
```

**Review by mondus — APPROVED** [review]
Approve but some optional changes that I think would help.

**Pull request merged**